.gitmodules

```diff
@@ -0,0 +1,6 @@
+[submodule "lib/cadmium"]
```

Comment:
no instructions on how to initiate and clone submodules. code breaks. doesn't run.

Reply:
adding the instructions in readme